crates/autopilot/src/run_loop.rs

```diff
@@ -148,48 +175,50 @@ impl RunLoop {
         }
     }
 
+    async fn update_caches(&self, prev_block: &mut Option<H256>, store_events: bool) -> BlockInfo {
```

Comment:
nit: I think `is_leader` is more appropriate. Currently the leader only stores events but if you think about it the leader could do any number of write operations in the future.
```suggestion
    async fn update_caches(&self, prev_block: &mut Option<H256>, is_leader: bool) -> BlockInfo {
```